axelor-account/src/main/java/com/axelor/apps/account/web/MoveLineQueryController.java

```diff
@@ -0,0 +1,135 @@
+        List<MoveLineQueryLine> moveLineQueryLineList = new ArrayList<MoveLineQueryLine>();
+        for (MoveLine moveLine : moveLineList) {
+          moveLineQueryLineList.add(new MoveLineQueryLine(moveLine, moveLineQuery));
+        }
+        response.setValue("moveLineQueryLineList", moveLineQueryLineList);
+      } else {
+        response.setValue("moveLineQueryLineList", null);
+      }
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void computeSelectedTotal(ActionRequest request, ActionResponse response) {
+    try {
+      MoveLineQuery moveLineQuery = request.getContext().asType(MoveLineQuery.class);
+      BigDecimal selectedCreditTotal = BigDecimal.ZERO;
+      BigDecimal selectedDebitTotal = BigDecimal.ZERO;
+      BigDecimal selectedBalanceTotal = BigDecimal.ZERO;
+
+      for (MoveLineQueryLine moveLineQueryLine : moveLineQuery.getMoveLineQueryLineList()) {
+        if (moveLineQueryLine.getIsSelected()) {
+          MoveLine moveLine = moveLineQueryLine.getMoveLine();
+          selectedCreditTotal = selectedCreditTotal.add(moveLine.getCredit());
+          selectedDebitTotal = selectedDebitTotal.add(moveLine.getDebit());
+        }
+      }
+      selectedBalanceTotal = selectedDebitTotal.subtract(selectedCreditTotal);
+
+      response.setValue("$selectedDebit", selectedDebitTotal);
+      response.setValue("$selectedCredit", selectedCreditTotal);
+      response.setValue("$selectedBalance", selectedBalanceTotal);
+
+      String balanceTitle;
+      if (selectedBalanceTotal.signum() > 0) {
+        balanceTitle = I18n.get("Debit balance");
+      } else if (selectedBalanceTotal.signum() < 0) {
+        balanceTitle = I18n.get("Credit balance");
+      } else {
+        balanceTitle = I18n.get("Balance");
+      }
+      response.setAttr("$selectedBalance", "title", balanceTitle);
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void massReconcile(ActionRequest request, ActionResponse response) {
+    try {
+      MoveLineQuery moveLineQuery = request.getContext().asType(MoveLineQuery.class);
+      moveLineQuery = Beans.get(MoveLineQueryRepository.class).find(moveLineQuery.getId());
+      List<MoveLine> moveLineList = new ArrayList<>();
+      if (!ObjectUtils.isEmpty(moveLineQuery.getMoveLineQueryLineList())) {
+        List<MoveLine> moveLineSelectedList =
+            moveLineQuery.getMoveLineQueryLineList().stream()
+                .filter(l -> l.getIsSelected())
+                .map(l -> l.getMoveLine())
+                .collect(Collectors.toList());
+        for (MoveLine moveLine : moveLineSelectedList) {
+          if (Beans.get(MoveLineControlService.class).canReconcile(moveLine)) {
+            moveLineList.add(moveLine);
+          }
+        }
+      }
+      if (!moveLineList.isEmpty()) {
+        Beans.get(MoveLineService.class).reconcileMoveLinesWithCacheManagement(moveLineList);
```

Comment:
Add a response.setReload(true) because we don't see any change otherwise. 
If you have time, improve the method MoveLineService.reconcileMoveLinesWithCacheManagement to return the number of reconciliations and display a message to inform the user about the number of reconciliations done.

Reply:
done without the message part